File: app/src/main/java/com/wadektech/eventshub/ui/FriendsEventsActivity.java
package com.wadektech.eventshub.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;

import com.wadektech.eventshub.R;
import com.wadektech.eventshub.adapter.FriendsEventsAdapter;
import com.wadektech.eventshub.models.FriendsEvents;
import com.wadektech.eventshub.viewmodels.FriendsEventsViewModel;

import java.util.ArrayList;
import java.util.List;

public class FriendsEventsActivity extends AppCompatActivity implements FriendsEventsAdapter.OnSingleFriendsEventItemClicked {
    RecyclerView mRecycler;
    LinearLayoutManager layoutManager;
    FriendsEventsAdapter fAdapter ;
    List<FriendsEvents> friendsEventsList ;
    FriendsEventsViewModel friendsEventsViewModel ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_friends_events);

        mRecycler = findViewById(R.id.rv_friends);
        layoutManager =  new LinearLayoutManager(this);
        mRecycler.setHasFixedSize(true);
        mRecycler.setLayoutManager(layoutManager);

        friendsEventsList = new ArrayList<>() ;

        friendsEventsViewModel = ViewModelProviders.of(this).get(FriendsEventsViewModel.class);
        friendsEventsViewModel.getAllFriendsEvents().observe(this, new Observer<List<FriendsEvents>>() {
            @Override
            public void onChanged(List<FriendsEvents> friendsEventsList) {
                fAdapter = new FriendsEventsAdapter(friendsEventsList, FriendsEventsActivity.this, FriendsEventsActivity.this::singleFriendsEventsItemClicked);
                mRecycler.setAdapter(fAdapter);
            }
        });
    }

    @Override
    public void singleFriendsEventsItemClicked(int position) {
        Intent intent = new Intent(FriendsEventsActivity.this, MoreInfo.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/wadektech/eventshub/ui/FriendsEventsActivity.kt
package com.wadektech.eventshub.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.wadektech.eventshub.R
import com.wadektech.eventshub.adapter.FriendsEventsAdapter
import com.wadektech.eventshub.adapter.FriendsEventsAdapter.OnSingleFriendsEventItemClicked
import com.wadektech.eventshub.database.MainEventsRoomDatabase
import com.wadektech.eventshub.repository.EventsHubRepository
import com.wadektech.eventshub.utils.EventsHubViewModelFactory
import com.wadektech.eventshub.viewmodels.EventsHubViewModel

class FriendsEventsActivity : AppCompatActivity(), OnSingleFriendsEventItemClicked {
    private lateinit var mRecycler: RecyclerView
    private lateinit var mLayout: LinearLayoutManager
    private lateinit var fAdapter: FriendsEventsAdapter
    private lateinit var eventsHubViewModel: EventsHubViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_friends_events)
        mRecycler = findViewById(R.id.rv_friends)

        initRecyclerview()

        fAdapter = FriendsEventsAdapter(this)
        mRecycler.adapter = fAdapter

        val db = MainEventsRoomDatabase(this)
        val repo = EventsHubRepository(db)
        val factory = EventsHubViewModelFactory(repo)

        eventsHubViewModel = ViewModelProvider(this, factory).get(EventsHubViewModel::class.java)
       eventsHubViewModel.getAllFriendsEvents.observe(this, Observer {
           fAdapter.submitList(it)
       })
    }

    override fun singleFriendsEventsItemClicked(position: Int) {
        val intent = Intent(this@FriendsEventsActivity, MoreInfo::class.java)
        startActivity(intent)
    }

    private fun initRecyclerview(){
        mRecycler.setHasFixedSize(true)
        mLayout = LinearLayoutManager(this)
        mRecycler.layoutManager = mLayout
    }
}